src/common/definitions.js

```diff
@@ -90,6 +91,86 @@ const address = (() => {
   };
 })();
 
+/** 
+ * Assembles schema for PCIU Addresses, which have properties and validations that differ from the standard
+ * common address definition. Note that this duplicates some code with 'address' common def, but want to
+ * ensure we're not tied to that so duplicating here will make things easy to delete if our approach changes.
+ * @returns {object} json-schema-form compatible schema object that conforms to PCIU address endpoint specs
+ */
+const pciuAddress = (() => {
+  const USA = 'USA';
+  // TODO: create custom field that will fetch countries / states (post-MVP)
+  const pciuCountryStateProperties = [{
+    // only need a state when country is 'USA'
+    properties: {
+      country: {
+        type: 'string',
+        enum: [USA]
+      },
+      state: {
+        type: 'string',
+        enum: constants.pciuStates
+      }
+    }
+  }, {
+    properties: {
+      country: {
+        type: 'string',
+        enum: constants.pciuCountries.filter(i => i !== USA)
+      }
+    }
+  }];
+  
+  // NOTE: Validations from swagger except where noted
+  return {
+    type: 'object',
+    oneOf: pciuCountryStateProperties,
+    required: ['addressLine1', 'country'],
+    properties: {
+      addressLine1: {
+        type: 'string',
+        maxLength: 35,
+        pattern: "([a-zA-Z0-9\-'.,,&#]([a-zA-Z0-9\-'.,,&# ])?)+$"
+      },
+      addressLine2: {
+        type: 'string',
+        maxLength: 35,
+        pattern: "([a-zA-Z0-9\-'.,,&#]([a-zA-Z0-9\-'.,,&# ])?)+$"
+      },
+      addressLine3: {
+        type: 'string',
+        maxLength: 35,
+        pattern: "([a-zA-Z0-9\-'.,,&#]([a-zA-Z0-9\-'.,,&# ])?)+$"
+      },
+      city: {
+        type: 'string',
+        maxLength: 35,
+        pattern: "([a-zA-Z0-9\-'.#]([a-zA-Z0-9\-'.# ])?)+$"
+      },
+      zipFirstFive: {
+        type: 'string',
+        pattern: '^\d{5}$' // not in swagger docs
+      },
+      zipLastFound: {
+        type: 'string',
+        pattern: '^\d{4}$' // not in swagger docs
+      },
+      militaryPostOfficeTypeCode: {
```

Comment:
So you're thinking you want to conditionally hide / require these in the `uiSchema`? I think that makes enough sense.

Reply:
I think so. Not having put together any uiSchemas, my opinion is liable to change at any time 🙃